.buildkite/hooks/pre-command

```diff
@@ -83,6 +83,33 @@ if [[ "${USE_PROD_DOCKER_CREDENTIALS:-}" == "true" ]]; then
 
   DOCKER_REGISTRY_PASSWORD="$(vault read -field=password secret/ci/elastic-elasticsearch/migrated/prod_docker_registry_credentials)"
   export DOCKER_REGISTRY_PASSWORD
+
```

Comment:
Can you just do `docker login --username "$DOCKER_REGISTRY_USERNAME" --password "$DOCKER_REGISTRY_PASSWORD" docker.elastic.co` instead of all of this stuff?

Reply:
(and by "can you" I mean: do you know if it works?)

Reply:
@brianseeders I tried that and the problem with that is, that we run this logic on all OS types, even those which don't have docker installed. using this file only approach allows not failing on those systems. Alternatively we could just not fail if docker is not available but that approach did not seem better than this file based one

Reply:
In that case, I would do:

```
if which docker > /dev/null 2>&1; then
  docker login --username "$DOCKER_REGISTRY_USERNAME" --password "$DOCKER_REGISTRY_PASSWORD" docker.elastic.co
fi
```

which has the advantage of not blowing away the existing docker config if there is one.

And probably move the vault lookups into the `if` as well, since they're useless on OSes that don't have docker, right?

Reply:
applied your suggestion

Reply:
I would approve the PR but you'll have to since I opened it